### PR TITLE
Enable tests against custom IoT Central endpoints

### PR DESCRIPTION
**Pull request by lucadruda**
With this changes, IoT Central integration tests can be run against internal clusters or different cloud.
Technically, it exposes the usual "--central-dns-suffix" as a configuration option for pytest.
No change in any commands.

- [ ] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [x] Have **all** unit **and** integration tests passed locally? i.e. `pytest <project root> -vv`
- [x] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.
- [ ] Have you made an entry in HISTORY.rst which concisely explains your feature or change?

image

### CODE REVIEW

File: azext_iot/tests/central/test_iot_central_int.py [hunk 1]
```diff
@@ -552,14 +539,14 @@ def test_central_device_registration_info_unassociated(self):
             == "Device does not have a valid template associated with it."
         )
 
-    @pytest.mark.skipif(
-        not DEVICE_ID, reason="empty azext_iot_central_primarykey env var"
+    @ pytest.mark.skipif(
```
Comment (digimaun) [inline azext_iot/tests/central/test_iot_central_int.py:542]: Unintended space?

File: azext_iot/tests/central/test_iot_central_int.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 APP_PRIMARY_KEY = os.environ.get("azext_iot_central_primarykey")
 APP_SCOPE_ID = os.environ.get("azext_iot_central_scope_id")
 DEVICE_ID = os.environ.get("azext_iot_central_device_id")
-TOKEN = os.environ.get("azext_iot_central_token")
+TOKEN = os.environ.get("azext_iot_central_token", None)
```
Comment (digimaun) [inline azext_iot/tests/central/test_iot_central_int.py:26]: `None` should be the default if the `azext_iot_central_token` key does not have a value.

File: azext_iot/central/providers/monitor_provider.py [hunk 3]
```diff
@@ -90,15 +91,15 @@ def _build_targets(
         return targets
 
     def _build_handler(
```
Comment (digimaun) [inline azext_iot/central/providers/monitor_provider.py:93]: Would recommend using python black for auto formatting.

File: azext_iot/tests/central/test_iot_central_int.py (hunk 1 above)
Comment (lucadruda) [reply azext_iot/tests/central/test_iot_central_int.py]: good catch. fixed

File: azext_iot/tests/central/test_iot_central_int.py (hunk 2 above)
Comment (lucadruda) [reply azext_iot/tests/central/test_iot_central_int.py]: removed

File: azext_iot/central/providers/monitor_provider.py (hunk 3 above)
Comment (lucadruda) [reply azext_iot/central/providers/monitor_provider.py]: changed to black and re-run formatting on the folder

**Review by digimaun — APPROVED** [review]

**Pull request merged**